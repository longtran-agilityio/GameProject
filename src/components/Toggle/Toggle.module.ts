import { Link } from 'react-router-dom'
import { styled, Box, Typography } from '@mui/material'
import { theme } from '@webapp/themes/index'

export const StyledLink = styled(Link)(({ theme }) => ({
  color: theme.palette.text.primary,
  fontSize: theme.typography.h6.fontSize,
  padding: theme.spacing(1.5, 9),
  margin: theme.spacing(0, 1),
  borderRadius: theme.spacing(0.2),
  border: 'none',
  cursor: 'pointer',
  textDecoration: 'none',
  textTransform: 'capitalize',
}))

export const StyledBox = styled(Box)(({ theme }) => ({
  display: 'flex',
  justifyContent: 'center',
  backgroundColor: theme.palette.background.paper,
  width: theme.spacing(52),
  padding: theme.spacing(0.5, 0),
  borderRadius: theme.spacing(0.2),
}))

export const StyledToggle = styled(Box)(({ theme }) => ({
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  paddingBottom: theme.spacing(10),
}))

export const TitleAccount = styled(Typography)(({ theme }) => ({
  paddingBottom: theme.spacing(5),
  fontWeight: '600',
  fontSize: theme.typography.h2.fontSize,
}))

export const StyledLinkActive = {
  backgroundColor: theme.palette.primary.main,
}
