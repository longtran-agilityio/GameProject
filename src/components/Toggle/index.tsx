// styles
import {
  StyledLink,
  StyledBox,
  StyledToggle,
  TitleAccount,
  StyledLinkActive,
} from './Toggle.module'

interface ILabel {
  currentUrl: string
  label: string
}

interface IToggle {
  labelToggle: ILabel[]
  pageUrl: string
  title: string
}

const Toggle = ({ title, pageUrl, labelToggle }: IToggle) => (
  <StyledToggle>
    <TitleAccount>{title}</TitleAccount>
    <StyledBox>
      {labelToggle.map((item) => (
        <StyledLink
          key={item.label}
          to={item.currentUrl}
          sx={{ ...(pageUrl === item.currentUrl && StyledLinkActive) }}
        >
          {item.label}
        </StyledLink>
      ))}
    </StyledBox>
  </StyledToggle>
)

export default Toggle
