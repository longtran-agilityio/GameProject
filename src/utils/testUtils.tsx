import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { ThemeProvider } from '@mui/material'
import { theme } from '@webapp/themes/index'

const AllBrowserRouter = ({ children }: { children: React.ReactNode }) => (
  <BrowserRouter>
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </BrowserRouter>
)

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllBrowserRouter, ...options })

export * from '@testing-library/react'
export { customRender as render }
