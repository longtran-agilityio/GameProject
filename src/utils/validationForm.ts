export const EmailValidation = {
  required: 'Email is required',
  pattern: {
    value: /^[\w-.]+@([\w-]+\.)+[\w-]{2,4}$/g,
    message: 'Invalid Email',
  },
}

export const PasswordValidation = {
  required: 'Password is required',
  minLength: { value: 8, message: 'min length is 8' },
}

export const FieldEmpty = {
  required: 'This field is required',
}
