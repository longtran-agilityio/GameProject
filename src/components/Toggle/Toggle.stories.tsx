// libs
import { ThemeProvider } from '@mui/material'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import { BrowserRouter } from 'react-router-dom'

// component
import Toggle from '.'

// theme
import { theme } from '@webapp/themes/index'

// mocks
import { labelToggle } from '@webapp/mocks/mockLabelToggle'

// constants
import { PageUrls } from '@webapp/constants/pageUrl'

export default {
  title: 'Components/Toggle',
  component: Toggle,
} as ComponentMeta<typeof Toggle>

const Template: ComponentStory<typeof Toggle> = (args) => (
  <BrowserRouter>
    <ThemeProvider theme={theme}>
      <Toggle {...args} />
    </ThemeProvider>
  </BrowserRouter>
)

export const ToggleLogin = Template.bind({})
ToggleLogin.args = {
  title: 'Account',
  labelToggle: labelToggle,
  pageUrl: PageUrls.LOGIN,
}

export const ToggleRegister = Template.bind({})
ToggleRegister.args = {
  title: 'Account',
  labelToggle: labelToggle,
  pageUrl: PageUrls.REGISTER,
}
