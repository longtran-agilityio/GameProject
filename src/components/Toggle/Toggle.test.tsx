import { render } from '@webapp/utils/testUtils'
import { faker } from '@faker-js/faker'

import Toggle from '.'

const ToggleProps = {
  pageUrl: faker.system.directoryPath(),
  title: faker.name.firstName(),
  labelToggle: [
    {
      currentUrl: faker.system.directoryPath(),
      label: faker.name.lastName(),
    },
  ],
}
const ToggleComponent = () =>
  render(
    <Toggle
      labelToggle={ToggleProps.labelToggle}
      pageUrl={ToggleProps.pageUrl}
      title={ToggleProps.title}
    />,
  )

describe('Toggle', () => {
  test('should matches snapshot', () => {
    const { container } = ToggleComponent()
    expect(container).toMatchSnapshot()
  })
  test('should render correctly', () => {
    const { getByText } = ToggleComponent()
    expect(getByText(ToggleProps.title)).toBeInTheDocument()
  })
})
