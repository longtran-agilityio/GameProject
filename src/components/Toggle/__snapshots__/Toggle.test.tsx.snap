// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle should matches snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-w0eck8"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ytizn3-MuiTypography-root"
    >
      Jude
    </p>
    <div
      class="MuiBox-root css-mrlnu"
    >
      <a
        class="css-17al1o8"
        href="/boot"
      >
        Feeney
      </a>
    </div>
  </div>
</div>
`;
